components/switch/src/Switch.js

```diff
@@ -62,14 +48,143 @@ const asArray = val => (Array.isArray(val) ? val : [val]);
 // ==============================
 // Component
 // ==============================
+export const Switch = ({
+	name,
+	label,
+	checked: isChecked,
+	onChange,
+	size,
+	block,
+	flipped,
+	toggleText,
+	disabled,
+	srOnlyText,
+}) => {
+	const { COLORS, [pkg.name]: localBrandTokens } = useBrand();
+	const mq = useMediaQuery();
+	const [checked, setChecked] = useState(isChecked);
+	const [switchId] = useState(`switch-${shortid.generate()}`);
+
+	const sizeArr = responsiveMap(asArray(size));
+
+	const localTokens = {
+		labelCss: {},
+		textCss: {},
+		toggleCss: {},
+		toggleTextCss: {},
+	};
+
+	merge(localTokens, localBrandTokens);
 
-export const Switch = ({ size, ...props }) => {
-	const flexiSize = responsiveMap(asArray(size));
+	useEffect(() => {
+		setChecked(isChecked);
+	}, [isChecked]);
+
+	const handleChange = name => () => {
+		if (onChange) {
+			onChange(name);
```

Comment:
We should use `wrapHandlers` from core and combine the consumer `onChange` with out internal state change.